pkg/controller/core/workload_controller.go

```diff
@@ -157,22 +159,22 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	}
 
 	if workload.SyncAdmittedCondition(&wl) {
-		if err := workload.ApplyAdmissionStatus(ctx, r.client, &wl, true); err != nil {
-			return ctrl.Result{}, err
-		}
-		if workload.IsAdmitted(&wl) {
+		//If it used to be admitted, don't update the status, trigger the eviction first
+		if !workload.IsAdmitted(&wl) {
```

Comment:
Can we maybe swap the order of checks - first `if workload.IsAdmitted(&wl) {`, so that we don't deal with double negation in else.